File: array-and-string/two-sum.scala
object Solution {
    def twoSum(numbers: Array[Int], target: Int): Array[Int] = {
        var (i, j) = (0, numbers.length - 1)
        while (i != j) {
            if (numbers(i) + numbers(j) < target)
                i += 1
            else if (numbers(i) + numbers(j) > target)
                j -= 1
            else
                return Array(i + 1, j + 1)
        }
        Array(i + 1, j + 1)
    }
}

File: hash-table/two-sum.cpp
#include <unordered_set>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashmap;
        for (int i = 0; i < nums.size(); i++) {
            if (hashmap.count(target - nums[i]))
                return {hashmap[target - nums[i]], i};
            hashmap[nums[i]] = i;
            // hashmap.insert(make_pair(nums[i], i));
        }
        return {};
    }
};